src/techui_builder/generate.py

```diff
@@ -157,6 +214,68 @@ def _create_widget(self, component: Entry) -> EmbeddedDisplay | ActionButton:
 
         return new_widget
 
+    def layout_widgets(self, widgets: list[EmbeddedDisplay | ActionButton]):
+        group_spacing = 30
+        max_group_height = 800
+        spacing_x = 20
+        spacing_y = 30
+        # Group tiles by size
+        groups = defaultdict(list)
+        for widget in widgets:
+            key = self._get_screen_dimensions_object(widget)
+
+            groups[key].append(widget)
+
+        # Sort groups by width (optional)
+        sorted_widgets = []
```

Comment:
This needs type-hinting
```suggestion
        sorted_widgets: list[EmbeddedDisplay | ActionButton] = []
```